src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -17,21 +18,90 @@ const SidebarSidecarLayout = ({
   sidebarProps,
   sidebarSlot,
   sidecarSlot,
+  sidebarPropsLevels,
 }: SidebarSidecarLayoutProps) => {
+  const { isDesktop } = useDeviceSize()
+  const numSidebarLevels = sidebarPropsLevels ? sidebarPropsLevels.length : 0
+  const initialSidebarLevel = useMemo(() => {
+    if (numSidebarLevels > 0) {
+      return numSidebarLevels - 1
+    } else {
+      return 0
+    }
+  }, [numSidebarLevels])
```

Comment:
Suggestion: `useMemo` feels redundant here, as deriving the value isn't computationally complex.

Reply:
This might be leftover from a slightly earlier iteration in the code... I had a warning that I should use `useMemo` because a value was changing in a way that caused and endless cycle of re-renders. I don't think that problem is here anymore, so I'll clean this up! Great catch 👀

Reply:
Removed! ✨